unlock-app/src/components/interface/checkout/main/Confirm.tsx

```diff
@@ -175,7 +181,7 @@ export function Confirm({
 
   const currencyContractAddress = swap
     ? payment.route.trade.inputAmount.currency?.address
-    : lock?.currencyContractAddress
+    : lock?.currencyContractAddress ?? undefined
```

Comment:
This is very confusing. Any reason why you set it to `undefined`? `lock?.currencyContractAddress` should already be undefined or null if unset

Reply:
Is passing the contract address -> `'null'` as a string, and this is not right